src/adam7.rs

```diff
@@ -6,6 +6,42 @@
 /// applies to a decoded row.
 ///
 /// See also [Reader.next_interlaced_row](crate::decoder::Reader::next_interlaced_row).
+use std::cmp::min;
+
+struct PassConstants {
+    line_mul: u8,
+    line_off: u8,
+    samp_mul: u8,
+    samp_off: u8,
+}
+
+impl PassConstants {
+    const fn new(line_mul: u8, line_off: u8, samp_mul: u8, samp_off: u8) -> Self {
+        Self {
+            line_mul,
+            line_off,
+            samp_mul,
+            samp_off,
+        }
+    }
+    const fn x_repeat(&self) -> u8 {
```

Comment:
nit: Could you please change the return value to `usize`?  (same for `fn y_repeat`)

* After changing the type of fields it shouldn't require changing the body of `fn x_repeat`
* It would avoid having to cast `as usize` in the only caller of `x_repeat()`.